Sources/SwiftMemcache/MemcachedValue.swift

```diff
@@ -0,0 +1,80 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the swift-memcache-gsoc open source project
+//
+// Copyright (c) 2023 Apple Inc. and the swift-memcache-gsoc project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of swift-memcache-gsoc project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIOCore
+
+/// Protocol defining the requirements for a type that can be converted to a ByteBuffer for transmission to Memcached.
+public protocol MemcachedValue {
+    /// Writes the value to a ByteBuffer.
+    ///
+    /// - Parameter buffer: The ByteBuffer to which the value should be written.
+    func writeToBuffer(_ buffer: inout ByteBuffer)
+
+    /// Reads the type from a ByteBuffer.
+    ///
+    /// - Parameter buffer: The ByteBuffer from which the value should be read.
+    static func readFromBuffer(_ buffer: inout ByteBuffer) -> Self?
+}
+
+/// Extension for FixedWidthInteger types to conform to MemcachedValue.
+extension MemcachedValue where Self: FixedWidthInteger {
+    /// Writes the integer to a ByteBuffer.
+    ///
+    /// - Parameter buffer: The ByteBuffer to which the integer should be written.
+    public func writeToBuffer(_ buffer: inout ByteBuffer) {
+        buffer.reserveCapacity(MemoryLayout<Self>.size)
```

Comment:
I don't think we have to do this. `writeInterger` is making sure that there is enough capacity before writing.